go/oasis-node/cmd/common/common.go

```diff
@@ -191,12 +193,21 @@ func GetInputReader(cmd *cobra.Command, cfg string) (io.ReadCloser, bool, error)
 }
 
 // LoadEntity loads the entity and it's signer.
-func LoadEntity(dataDir string) (*entity.Entity, signature.Signer, error) {
+func LoadEntity(backing string) (*entity.Entity, signature.Signer, error) {
 	if flags.DebugTestEntity() {
 		return entity.TestEntity()
 	}
 
 	// TODO/hsm: Configure factory dynamically.
-	entitySignerFactory := fileSigner.NewFactory(dataDir, signature.SignerEntity)
-	return entity.Load(dataDir, entitySignerFactory)
+	switch backing {
+	case ledgerSigner.SignerName:
+		factory := ledgerSigner.NewFactory(flags.EntityFileDir(), signature.SignerEntity)
+		signer, err := factory.Load(signature.SignerEntity)
+		return nil, signer, err
+	case fileSigner.SignerName:
+		factory := factorySigner.NewFactory(flags.EntityLedgerAddress(), signature.SignerEntity)
+		return entity.Load(dataDir, factory)
```

Comment:
Where does the `dataDir` come from? You should call `DataDir()` or `DataDirOrPwd()` here.